views/cart.php

```diff
@@ -0,0 +1,116 @@
+<div class="cart__container">
+    <h4>Your Cart</h4>
+    <div class="cart__table">
+        <table id="cartTable">
+            <tr>
+                <th></th>
+                <th>Product</th>
+                <th>Price</th>
+                <th>Quantity</th>
+                <th>Total</th>
+            </tr>
+
+            <tr>
+                <td class="cart__item__remove">
+                    <i class="far fa-trash-alt"></i>
+                </td>
+                <td>
+                    <img src="./assets/Images/background.jpg" alt="">
+                    Product 1
+                </td>
+                <td>000.000.000</td>
+                <td class="cart__item__quantity">
+                    <input type="number" value="1" step="1" min="1" max="100" class="input-group">
+                </td>
+                <td class="cart__item__total">000.000.000</td>
+            </tr>
+            <tr>
+                <td class="cart__item__remove">
+                    <i class="far fa-trash-alt"></i>
+                </td>
+                <td>
+                    <img src="./assets/Images/background.jpg" alt="">
+                    Product 1
+                </td>
+                <td>000.000.000</td>
+                <td class="cart__item__quantity">
+                    <input type="number" value="1" step="1" min="1" max="100" class="input-group">
+                </td>
+                <td class="cart__item__total">000.000.000</td>
+            </tr>
+            <tr>
+                <td class="cart__item__remove">
+                    <i class="far fa-trash-alt"></i>
+                </td>
+                <td class="cart__item__name">
+                    <img src="./assets/Images/background.jpg" alt="">
+                    Product 1
+                </td>
+                <td>000.000.000</td>
+                <td class="cart__item__quantity">
+                    <input type="number" value="1" step="1" min="1" max="100" class="input-group">
+                </td>
+                <td class="cart__item__total">000.000.000</td>
+            </tr>
+        </table>
+    </div>
+</div>
+<script>
+    $(function() {
```

Comment:
I highly recommend moving away from jQuery. This was technology that was used at one time to normalize browser behaviors. Most modern browsers now behave quite similarly, so the need for jQuery is no longer there.

Reply:
So I should use other libraries like Vue, React, Angular or perhaps, vanilla JS? This is where it gets fuzzy to me because I've yet to catch up on the Frontend developement.

Reply:
In this particular case, since it's just javascript on an HTML page that is being served, I would recommend vanilla JS. 

However, if you are getting into a frontend framework, I would stay away from Angular. Either React or Vue would be good. It is better to use a modern framework than to use a library like jQuery. jQuery was useful at one time due to its flexibility, but that flexibility results in poor code hygene.

Reply:
So I have a page's contents placeholder within the `main.html` layout file as `{{content}}`. And by using vanilla JS, I'm now facing issues on DOM of which I can't get values of dynamically created elements. The reason behind this, I think, is that the script files are isn't call at the end of the body tag. (In the layout, there's `navbar` > `{{content}}` > `footer`). I guess I'll have to create a `{{scripts}}` placeholder at the end and load them respectively...

Reply:
```js
$(function() {
  console.log('hi');
});
```

In jQuery, this is a shorthand for executing when the DOM is ready. In essence, it is hard to mimic that functionality. However, since javascript is single-threaded, try the following to see if it works:

```js
setTimeout(function() {
  console.log('hi');
}, 0);
```

Note the second parameter of `setTimeout` is `0`.

This is exactly the reason why you should not use this pattern at all. You are introducing javascript that is executed when it is written to the page, this is really hard to debug if there's something wrong with the code.

Reply:
Sorry, I missed this question earlier. If you don't see a response, feel free to let me know in slack.

Reply:
So it is bad to have scripts executing at page loading? And what practice should I follow? I also read about page pre-loading.